storage-fetch/src/download.rs

```diff
@@ -172,7 +201,10 @@ impl Client {
 
             FileReader::new(file)
                 .await?
-                .copy_tree(&self.root, &mut extracted_file)
+                .copy_tree(
+                    &self.root.expect("root should be known"),
```

Comment:
Maybe you could check the root with a `let` guard and return a WrongNumberOfRoots or similar